### ISSUE TITLE
not getting a single one bounding box

### ISSUE THREAD

**Issue by GabrielGlzSa** [yhenon/keras-frcnn#14]
Hi, first of all thank you for sharing your code.

I wanna use your code to train a dataset with only 4 categories for detecting one type of obstacle and 3 marks, but first I trained the net with PASCAL VOC to try it out. When I tested the net, none of the photos I had tried had a bounding box, including 110 photos I took from the dataset.

Would you happen to know why no bounding box is appended?

Thanks in advance.
```
for ii in range(P_cls.shape[1]):
			
			if np.max(P_cls[0,ii,:]) < 0.5 or np.argmax(P_cls[0,ii,:]) == (P_cls.shape[2] - 1):
				continue

			cls_name = class_mapping[np.argmax(P_cls[0,ii,:])]

			if cls_name not in bboxes:
				bboxes[cls_name] = []
				probs[cls_name] = []

			(x, y, w, h) = ROIs[0,ii,:]

			cls_num = np.argmax(P_cls[0, ii, :])
			(tx, ty, tw, th) = P_regr[0, ii, 4*cls_num:4*(cls_num+1)]
			x, y, w, h = roi_helpers.apply_regr(x, y, w, h, tx, ty, tw, th)

			bboxes[cls_name].append([16*x, 16*y, 16*(x+w), 16*(y+h)])
			probs[cls_name].append(np.max(P_cls[0, ii, :]))
```

**Comment by small-yellow-duck** [comment]
Can you check that your test images correctly normalized and that the colour channels are in the right order when you make a prediction for them?

**Comment by yassinzak** [comment]
I checked on the color channel and normalization  but still not working . There is no bounding box result at all. I use tensorflow and trained it for 50 epoch.  There is results coming from RPN which i'm still not sure if it works correctly or not or if there is an error that i have missed but the classifier always gives  background classification.

**Comment by yhenon** [comment]
Could you share the training log please?

**Comment by GabrielGlzSa** [comment]
Do you mean the hdf5 file?
If not, where can I see the training log?

I left my computer training for a couple of days while I was on vacation.

nb_epochs = 300
train_samples_per_epoch = 500

It seems that increasing epochs fixed it a bit cause now I can see only two categories: person and airplane. However, it is detecting airplane when there isn't a single one in the image. At least now I can see a bounding box in the image. 
Here is my model:
https://drive.google.com/open?id=0B5Pc0luoI--_dk1Vc19CZjVpQzQ

Thanks in advance for your help :)

**Comment by yassinzak** [comment]
I removed the comment  from this part of the code  https://github.com/yhenon/keras-frcnn/blob/master/keras_frcnn/data_generators.py   line 255  and it begin to give results a better one. with high recall and low precision on custom dataset .

**Comment by yhenon** [comment]
@yassinzak is correct that the section of code had been accidentally removed. The repo has been updated with this issue fixed. Also, now when training the code reports whether the RPN is succesfully predicting bounding boxes, and the % accuracy of the classifier on boxes.

**Comment by ant1pink** [comment]
I have the same issue

**Comment by Abhijit-2592** [comment]
@ant1pink  try this [repo](https://github.com/Abhijit-2592/Keras_object_detection). I have modified this repo and fixed some bugs. Lemme know if the problem still persists

**Comment by ant1pink** [comment]
The problem I got is the code never passed the following sentence:
if np.max(P_cls[0, ii, :]) < final_classification_threshold or np.argmax(P_cls[0, ii, :]) == (P_cls.shape[2] - 1):
    continue

I will try the new one.

Cheers
